server/server.go

```diff
@@ -0,0 +1,198 @@
+package server
+
+import (
+	"context"
+	"errors"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/tendermint/iavl"
+	pb "github.com/tendermint/iavl/proto"
+)
+
+var _ pb.IAVLServiceServer = (*IAVLServer)(nil)
+
+// iavlServer implements the gRPC IAVLServiceServer interface. It provides a gRPC
+// API over an IAVL tree.
+type IAVLServer struct {
+	tree *iavl.MutableTree
+}
+
+func New(db dbm.DB, cacheSize, version int64) (*IAVLServer, error) {
+	tree := iavl.NewMutableTree(db, int(cacheSize))
+
+	// TODO: Consider storing (per version save) and load latest height when the
```

Comment:
@alexanderbez Quick question on your `TODO` here. I like the idea of defaulting to the latest height when `version` is 0. I don't quite follow the `Consider storing (per version save)` part, could you elaborate some more?

Reply:
Actually it looks like that is the default behavior anyway:
https://github.com/tendermint/iavl/blob/master/mutable_tree.go#L243-L246

Reply:
> I took this to mean somehow looking up the latest version from the DB.

Yes, exactly. This is correct. This is precisely what the SDK does: https://github.com/cosmos/cosmos-sdk/blob/3c2ace95100d47f7fc4f68241fa91c2158ee368a/store/rootmulti/store.go#L596. You'll notice we save the version (on commit) under a unique key. The TODO suggests we do the same here.

The reason it is a TODO is because idk if we _should_ do this or if we should keep the API and "clean" as possible. The latter would mean the client would need to know the latest height.

Reply:
@alexanderbez Ahh okay, I get you now. Since the sdk takes responsibility for saving the version I'd vote that we leave it out of the gRPC server as it then better reflects the IAVL native API.